packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -107,43 +118,97 @@ contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDoma
         emit SetDepositContract(depositContract);
     }
 
+    /**************************************************
+     *        CROSSDOMAIN ADMIN FUNCTIONS             *
+     **************************************************/
+
+    /**
+     * @notice Set this contract as the admin address in the L2 Deposit contract.
+     * @dev Only callable by the current owner.
+     * @param _l2Gas Gas limit to set for relayed message on L2
```

Comment:
I feel that this param (used in a few other places as well) is pretty fragile. I've tested out what happens if this is too low or too high and it can fail pretty badly. If it's too low (you set it to 0 for example) then the L1 tx will go through but nothing happens on L2 (gas too low). if it's too high (larger than 12.5m, optimism gas limit) it reverts on L1 due to the cross-domain messenger.

It might be worthwhile to set a variable such as 10m that we always use for the l2Gas in all calls to prevent the caller from messing this up. wdyt?

Reply:
hmmm yeah that's reasonable, we should actually ask the Optimism team. I see that [makerdao](https://github.com/makerdao/optimism-dai-bridge/blob/master/contracts/l1/L1DAITokenBridge.sol#L84) leaves this as a parameter so i'm curious what best practices are.

Reply:
I think we can leave it as a param for now. will ask the Optimism team about the best practices.